packages/engine/src/scene/systems/ShadowSystem.tsx

```diff
@@ -302,6 +302,15 @@ const execute = () => {
 }
 
 const reactor = () => {
+  const rendererState = getMutableState(RendererState)
+  const useShadows = useHookstate(rendererState.useShadows)
+  const renderMode = useHookstate(rendererState.renderMode)
+  const isEditor = useHookstate(getMutableState(EngineState).isEditor)
+  const sessionState = useHookstate(getMutableState(XRState).sessionActive)
+  useEffect(() => {
+    EngineRenderer.instance.renderer.shadowMap.enabled = getShadowsEnabled()
```

Comment:
we can remove the redundant `getShadowsEnabled()` here and combine these set operations

Reply:
I thought about it further and also think this doesn't have to be reactive. You're either a mobile XR device, IOS, editor, or not. I don't believe i've seen us swapping shadow render modes.